pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go

```diff
@@ -75,94 +75,144 @@ type ReconcilePerconaServerMongoDBRestore struct {
 
 // Reconcile reads that state of the cluster for a PerconaServerMongoDBRestore object and makes changes based on the state read
 // and what is in the PerconaServerMongoDBRestore.Spec
-// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
-// a Pod as an example
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcilePerconaServerMongoDBRestore) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	rr := reconcile.Result{
+		RequeueAfter: time.Second * 5,
+	}
+
 	// Fetch the PerconaSMDBBackupRestore instance
 	instance := &psmdbv1.PerconaServerMongoDBRestore{}
 	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if k8serrors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			return reconcile.Result{}, nil
+			return rr, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		return rr, err
 	}
 
 	err = instance.CheckFields()
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("fields check: %v", err)
+		return rr, fmt.Errorf("fields check: %v", err)
 	}
 
 	if instance.Status.State == psmdbv1.RestoreStateReady {
-		return reconcile.Result{}, nil
+		return rr, nil
 	}
 
 	err = r.reconcileRestore(instance)
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("reconcile: %v", err)
+		return rr, fmt.Errorf("reconcile: %v", err)
 	}
 
-	return reconcile.Result{}, nil
+	return rr, nil
 }
 
-func (r *ReconcilePerconaServerMongoDBRestore) reconcileRestore(cr *psmdbv1.PerconaServerMongoDBRestore) error {
-	backup, err := r.getBackup(cr)
+func (r *ReconcilePerconaServerMongoDBRestore) reconcileRestore(cr *psmdbv1.PerconaServerMongoDBRestore) (err error) {
+	status := cr.Status
+
+	defer func() {
```

Comment:
I don't understand the question. Defer is run always no matter what happened earlier. If there is an error it will set respective state. And if the state has changed it will update it in k8s.

Reply:
in other words -  should we run the status check for finished backups?

Reply:
We won't because of https://github.com/percona/percona-server-mongodb-operator/blob/e3e40a6c9fc9dc7ba817498a2dba5f299ff59521/pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go#L104-L106

But I'll add Error state to that clause as well.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/312/files/ba1257438f7d594ddc36d219c8476553ffaedab3..823e637a1e93e9f70a135ee889a87636433e1b7c#diff-6c58e37e9daa9cb44bfab0dc0b53cddaR106